src/DocumentDB.ChangeFeedProcessor/LeaseManagement/DocumentServiceLease.cs

```diff
@@ -90,6 +92,19 @@ public static DocumentServiceLease FromDocument(Document document)
             return JsonConvert.DeserializeObject<DocumentServiceLease>(json);
         }
 
+        public static object StampWithCustomPartitionKeyProperty(ILease lease, string customPartitionKeyPropertyName)
+        {
+            object leaseDoc = lease;
+            if (!customPartitionKeyPropertyName.Equals(DocumentServiceLease.IdPropertyName, StringComparison.OrdinalIgnoreCase))
+            {
+                var leaseJDoc = JObject.FromObject(lease);
+                leaseJDoc.Add(customPartitionKeyPropertyName, lease.Id);
```

Comment:
Would it work if this used new Guid() as the value? Ideally, we'd like equal distribution of the leases across physical partitions.

Reply:
since we already use Id as partition key, using the same value guarantees that the distribution is not regressing.
since in this PR, we are not touching ILease interface to have an explicit partition key property. when we query, there is no way for us to know what is the partition key of the underlying lease.